python/pyarrow/tests/test_parquet.py

```diff
@@ -2169,13 +2303,15 @@ def read_multiple_files(paths, columns=None, use_threads=True, **kwargs):
     assert result.equals(expected)
 
     # Read with provided metadata
-    metadata = pq.read_metadata(paths[0])
+    # TODO(dataset) specifying metadata not yet supported
```

Comment:
We'll have to discuss how to expose metadata in `pyarrow.dataset`. IMO this should be a property of ParquetFileFragment and not of a Dataset

Reply:
> We'll have to discuss how to expose metadata in pyarrow.dataset. IMO this should be a property of ParquetFileFragment and not of a Dataset

Yes, indeed, this could be something specific on the ParquetFileFragment, although a ParquetDataset can still have a "metadata" that maps to all metadata of all files/row groups, *if available* (eg if the dataset was created from a `_metadata` file with all this information). 

And with the future work on storing statistics on fragments, we might also get a parquet-independent access to part of the metadata.

Reply:
`_metadata` serves as a consolidated mapping from paths or row groups to metadata, so I'd say that even in that case the metadata derived from it is a per-Fragment property.